plugins/source/azure/codegen/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"runtime"
+	"sort"
+	"strings"
+	"text/template"
+
+	codegen "github.com/cloudquery/cloudquery/plugins/source/azure/codegen/recipes"
+	sdkgen "github.com/cloudquery/plugin-sdk/codegen"
+	"github.com/gertd/go-pluralize"
+	"github.com/iancoleman/strcase"
+	"golang.org/x/exp/maps"
+)
+
+//go:embed templates/*.go.tpl
+var azureTemplatesFS embed.FS
+
+type providerResource struct {
+	ServiceName string
+	Function    string
+}
+
+type provider struct {
+	Packages  []string
+	Resources []providerResource
+}
+
+func main() {
+	filter := ""
+	if len(os.Args) > 1 {
+		filter = os.Args[1]
+	}
+	pattern := regexp.MustCompile("(?i)" + filter)
+	resourcesDir := path.Join(path.Dir(getFilename()), "../resources")
+	providerDir := path.Join(resourcesDir, "provider")
+	servicesDir := path.Join(resourcesDir, "services")
+
+	providerResources := make(map[string]providerResource)
+	providerPackages := make(map[string]bool)
+
+	for _, r := range codegen.AllResources() {
```

Comment:
I tried to keep this file as simple as possible and leave all  resource creating logic under `recipes/base.go`